### DISCUSSION TITLE
What is the efficient way to set the password rules before the insert to DB?

### DISCUSSION THREAD

**Discussion by cdpython** [supabase/supabase#1592] [Questions] [answered]
Thank you for sharing the awesome project.

I wonder how to implement and apply the password rules when users create their accounts.

for example "AWS Cognito" has password restrict features such as a minimum length of the password, include Special characters, and so on before insert to DB.

My plan is to use the Cloudflare Worker Proxy when the account signup process.

Is there an efficient way to implement that?

Thank you.

**Comment by soedirgo** [comment, marked as answer]
If you can do this outside the DB, you can try something like [zxcvbn](https://github.com/dropbox/zxcvbn) (or its newer fork [zxcvbn-ts](https://github.com/zxcvbn-ts/zxcvbn)) to estimate password strength.

**Reply by cdpython** [reply]
Thank you! Looks like it can be used in 'Cloudflare Worker'!
